src/v/cloud_storage/remote_partition.cc

```diff
@@ -364,6 +373,38 @@ ss::future<> remote_partition::start() {
     co_return;
 }
 
+void remote_partition::bg_prefetch_segment(iterator it) {
+    it++;
+    if (it == end()) {
+        return;
+    }
+    ssx::spawn_with_gate(_gate, [this, it] {
```

Comment:
Before we decide to do this, is there a way we can check how heavily loaded the cache is?  In the scenario where we have e.g. 128MB segments, 1000 partitions, and a 128GB cache, we only have space for each partition to hydrate one segment at a time, so probably nobody should do any prefetching.

Not sure what the right heuristic is, but feels like we need something.